Libs/MRML/Core/vtkMRMLTextNode.h

```diff
@@ -0,0 +1,96 @@
+/*=auto=========================================================================
+
+  Portions (c) Copyright 2009 Brigham and Women's Hospital (BWH) All Rights Reserved.
+
+  See Doc/copyright/copyright.txt
+  or http://www.slicer.org/copyright/copyright.txt for details.
+
+  Program:   3D Slicer
+  Module:    $RCSfile: vtkMRMLTextNode.h,v $
+  Date:      $Date: 2006/03/19 17:12:29 $
+  Version:   $Revision: 1.3 $
+
+=========================================================================auto=*/
+
+#ifndef __vtkMRMLTextNode_h
+#define __vtkMRMLTextNode_h
+
+// MRML includes
+#include <vtkMRMLStorableNode.h>
+
+// STD includes
+#include <sstream>
+
+class  VTK_MRML_EXPORT vtkMRMLTextNode : public vtkMRMLStorableNode
+{
+public:
+
+  static vtkMRMLTextNode* New();
+  vtkTypeMacro(vtkMRMLTextNode, vtkMRMLStorableNode);
+  void PrintSelf(ostream& os, vtkIndent indent) override;
+
+  virtual vtkMRMLNode* CreateNodeInstance() override;
+
+  ///
+  /// Set node attributes
+  virtual void ReadXMLAttributes(const char** atts) override;
+
+  ///
+  /// Write this node's information to a MRML file in XML format.
+  virtual void WriteXML(ostream& of, int indent) override;
+
+  ///
+  /// Copy the node's attributes to this object
+  virtual void Copy(vtkMRMLNode* node) override;
+
+  ///
+  /// Get node XML tag name (like Volume, Model)
+  virtual const char* GetNodeTagName() override { return "Text"; };
+
+  /// Set text node contents
+  void SetText(const std::string text, int encoding=VTK_ENCODING_NONE);
```

Comment:
It is interesting that this compiled (const is in the header but not in the implementation). Anyway, it probably should be const std::string&.